### ISSUE TITLE
building error

### ISSUE THREAD

**Issue by GregTroar** [autobrr/autobrr#42]
Hello,

I tried to use the binaries on my seedbox and everything loads up as expected except that when I try to create an indexer the dropdown list is empty and then I get an error so I tried to build the binaries myself and here is the error I get at the last line:

> go: downloading github.com/asaskevich/EventBus v0.0.0-20200907212545-49d423059eef
go: downloading github.com/r3labs/sse/v2 v2.7.2
go: downloading github.com/rs/zerolog v1.26.0
go: downloading github.com/spf13/pflag v1.0.5
go: downloading modernc.org/sqlite v1.14.1
go: downloading github.com/spf13/viper v1.9.0
go: downloading github.com/gdm85/go-libdeluge v0.5.5
go: downloading github.com/Masterminds/squirrel v1.5.1
go: downloading github.com/lib/pq v1.10.4
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/anacrolix/torrent v1.38.0
go: downloading gopkg.in/irc.v3 v3.1.4
go: downloading gopkg.in/natefinch/lumberjack.v2 v2.0.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/go-chi/chi v1.5.4
go: downloading github.com/gorilla/sessions v1.2.1
go: downloading github.com/rs/cors v1.8.0
go: downloading golang.org/x/crypto v0.0.0-20211117183948-ae814b36b871
go: downloading golang.org/x/net v0.0.0-20211123203042-d83791d6bcd9
go: downloading gopkg.in/cenkalti/backoff.v1 v1.1.0
go: downloading github.com/fsnotify/fsnotify v1.5.1
go: downloading github.com/magiconair/properties v1.8.5
go: downloading github.com/mitchellh/mapstructure v1.4.2
go: downloading github.com/spf13/afero v1.6.0
go: downloading github.com/spf13/cast v1.4.1
go: downloading github.com/spf13/jwalterweatherman v1.1.0
go: downloading github.com/subosito/gotenv v1.2.0
go: downloading gopkg.in/ini.v1 v1.64.0
go: downloading github.com/lann/builder v0.0.0-20180802200727-47ae307949d0
go: downloading github.com/gdm85/go-rencode v0.1.8
go: downloading github.com/anacrolix/missinggo v1.3.0
go: downloading github.com/gorilla/securecookie v1.1.1
go: downloading golang.org/x/sys v0.0.0-20211124211545-fe61309f8881
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/pelletier/go-toml v1.9.4
go: downloading golang.org/x/text v0.3.7
go: downloading github.com/lann/ps v0.0.0-20150810152359-62de8c46ede0
go: downloading github.com/bradfitz/iter v0.0.0-20191230175014-e8f45d346db8
go: downloading github.com/anacrolix/missinggo/v2 v2.5.2
go: downloading github.com/huandu/xstrings v1.3.2
go: downloading modernc.org/libc v1.11.82
go: downloading github.com/google/uuid v1.3.0
go: downloading github.com/mattn/go-isatty v0.0.14
go: downloading modernc.org/mathutil v1.4.1
go: downloading modernc.org/memory v1.0.5
go: downloading github.com/remyoudompheng/bigfft v0.0.0-20200410134404-eec4a21b6bb0
../../web/build.go:14:12: pattern build: no matching files found

Any idea ?

Thanks

**Comment by ludviglundgren** [comment]
Hi. 

If you look in the Dockerfile how things are built, it first needs to build the frontend in `web`.

1. Make sure you have node/npm/yarn installed
2. navigate into `web`
3. Install packages with `npm install` or `yarn`
4. Build `npm run build` or `yarn build`
5. Then you can go back up into the root and build the binary

Again, more official instructions ARE coming.

**Comment by rouggy** [comment]
Awesome let me try
By the way with the binaries here is the issue I have:

[https://imgur.com/lC1L1Ml](url)

And i am running:

Distributor ID: Debian
Description:    Debian GNU/Linux 11 (bullseye)
Release:        11
Codename:       bullseye

after this error if I refresh I am redirected to login screen

**Comment by ludviglundgren** [comment]
Here's how Swizzin (seedbox installer/distro/package manager) sets everything up. [swizzin autobrr install](https://github.com/swizzin/swizzin/blob/master/scripts/install/autobrr.sh)

It uses a few helpers etc for nginx, but most is in there. Downloads binary, unpacks, moves, creates a systemd service file, creates config in the expected place in users home, then creates the initial user with `autobrrctl`.

**Comment by rouggy** [comment]
I achieved to build the binaries, unfortunately on my seedbox it still does not work with same error, I will run my tests on a local VM.

Nothing in the logs, also something strange, I added 3 definitions thus 18 indexers but it counts only 17 in the trace log

> 2021-12-21T16:24:48+01:00 DBG Starting autobrr
2021-12-21T16:24:48+01:00 DBG Log-level: TRACE
2021-12-21T16:24:48+01:00 INF Starting server. Listening on 0.0.0.0:8989
2021-12-21T16:24:48+01:00 TRC parsing: definitions/alpharatio.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/beyondhd.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/btn.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/emp.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/filelist.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/gazellegames.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/hd-torrents.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/iptorrents.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/milkie.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/nebulance.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/orpheus.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/ptp.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/red.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/superbits.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/torrentDay.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/torrentSeeds.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/torrentleech.yaml
2021-12-21T16:24:48+01:00 TRC parsing: definitions/uhdbits.yaml
2021-12-21T16:24:48+01:00 INF Loaded 17 indexer definitions

**Comment by rouggy** [comment]
Anyway this issue is resolved, I will try to further investigate why it works well on Debian 11 on my VM and not on my seedbox which also runs Debian 11.

**Comment by ludviglundgren** [comment]
That's indeed weird the numbers don't match. There's plenty of logging for the parsing so that should catch any errors. I see your fork with milkie and TD, could you push TS so I could check for any errors? You're more than welcome to make a PR with the additions as well. But please change to some non obvious release names before doing so, like the current definitions have. :) 

Could it be a browser cache issue on the web? Try and clear/remove the database as well if you have started it with different results. It should run the migrations when you set up the user with `autobrrctl`.

And could you please maybe continue on this issue for further discussion if needed? I'll close the others but keep this open for a couple days.

**Comment by rouggy** [comment]
Hi Ludvig,

I am still scratching my head on this issue.
I have deleted the database and config file, recreated user with autobrrctl --config /path/ create-user test
I have cleared all the chrome cache, changed the binaries to directory /tmp and chmoded 777 for testing purpose.
It does not work.

Version of linux on my seedbox is Linux 5.10.0-9-amd64 #1 SMP Debian 5.10.70-1 (2021-09-30) x86_64 GNU/Linux and is exactly the same on my virtual machine...

Problem is I have no log to look out to find out the issue :(

I have done 3 indexers and have updated the the releases names to something non obvious. I will test them and need to read how to do a PR cause I have never done any :)
